### DISCUSSION TITLE
STM32LoRaWAN mitake with WL55 chip

### DISCUSSION THREAD

**Discussion by ebelouet** [stm32duino/Arduino_Core_STM32#2208] [Libraries] [answered]
HI,
Why I have this mistake  with                   👍 

C:\Users\Utilisateur\Documents\Arduino\libraries\STM32LoRaWAN-main\src\BSP\radio_board_if.c:75:33: error: 'PC3' undeclared here (not in a function); did you mean 'PC_3'?
   75 |   #define LORAWAN_RFSWITCH_PINS PC3,PC4,PC5
      |                                 ^~~
C:\Users\Utilisateur\Documents\Arduino\libraries\STM32LoRaWAN-main\src\BSP\radio_board_if.c:84:70: note: in expansion of macro 'LORAWAN_RFSWITCH_PINS'
   84 | static unsigned lorawan_rfswitch_pins[LORAWAN_RFSWITCH_PIN_COUNT] = {LORAWAN_RFSWITCH_PINS};
      |                                                                      ^~~~~~~~~~~~~~~~~~~~~
C:\Users\Utilisateur\Documents\Arduino\libraries\STM32LoRaWAN-main\src\BSP\radio_board_if.c:75:37: error: 'PC4' undeclared here (not in a function); did you mean 'PC_4'?
   75 |   #define LORAWAN_RFSWITCH_PINS PC3,PC4,PC5
      |                                     ^~~
C:\Users\Utilisateur\Documents\Arduino\libraries\STM32LoRaWAN-main\src\BSP\radio_board_if.c:84:70: note: in expansion of macro 'LORAWAN_RFSWITCH_PINS'
   84 | static unsigned lorawan_rfswitch_pins[LORAWAN_RFSWITCH_PIN_COUNT] = {LORAWAN_RFSWITCH_PINS};
      |                                                                      ^~~~~~~~~~~~~~~~~~~~~
C:\Users\Utilisateur\Documents\Arduino\libraries\STM32LoRaWAN-main\src\BSP\radio_board_if.c:75:41: error: 'PC5' undeclared here (not in a function); did you mean 'PC_5'?
   75 |   #define LORAWAN_RFSWITCH_PINS PC3,PC4,PC5
      |                                         ^~~
C:\Users\Utilisateur\Documents\Arduino\libraries\STM32LoRaWAN-main\src\BSP\radio_board_if.c:84:70: note: in expansion of macro 'LORAWAN_RFSWITCH_PINS'
   84 | static unsigned lorawan_rfswitch_pins[LORAWAN_RFSWITCH_PIN_COUNT] = {LORAWAN_RFSWITCH_PINS};

Sincerely
Eric

**Comment by fpistm** [comment]
Which board you used? Which core version ?

**Reply by ebelouet** [reply]
stm32WL generic and generic WLE5C.

In fact I have STM32WLE5CCU6 chip.

**Reply by fpistm** [reply, marked as answer]
Hi @ebelouet 
You have to define the correct board configuration.
The library defines a default one for the Nucleo WL55JC1 as stated here:
https://github.com/stm32duino/STM32LoRaWAN/blob/192c1234a8661f72f1635e5675db0d11fb110920/src/BSP/radio_board_if.c#L41-L81

Example for RAK module:
https://github.com/stm32duino/Arduino_Core_STM32/blob/cd3043f3046fcbfafca6af99b2af26f5ab6fa9dc/variants/STM32WLxx/WL54CCU_WL55CCU_WLE4C(8-B-C)U_WLE5C(8-B-C)U/variant_RAK3172_MODULE.h#L146-L156

So you have to define those definitions to be able to build and run properly.
You can add an file named `hal_conf_extra.h` and add your configuration, ex:

```
// LoRaWAN definitions
#define LORAWAN_BOARD_HAS_TCXO          0U
#define LORAWAN_BOARD_HAS_DCDC          1U
#define LORAWAN_TX_CONFIG               RBI_CONF_RFO_HP

#define LORAWAN_RFSWITCH_PINS           PB8, PC13
#define LORAWAN_RFSWITCH_PIN_COUNT      2
#define LORAWAN_RFSWITCH_OFF_VALUES     LOW, LOW
#define LORAWAN_RFSWITCH_RX_VALUES      HIGH, LOW
#define LORAWAN_RFSWITCH_RFO_LP_VALUES  LOW, HIGH
#define LORAWAN_RFSWITCH_RFO_HP_VALUES  LOW, HIGH
```

or a file named `build_opt.h` with:
```
-DLORAWAN_BOARD_HAS_TCXO=0U
-DLORAWAN_BOARD_HAS_DCDC=1U
-DLORAWAN_TX_CONFIG=RBI_CONF_RFO_HP
-DLORAWAN_RFSWITCH_PINS=PB8,PC13
-DLORAWAN_RFSWITCH_PIN_COUNT=2
-DLORAWAN_RFSWITCH_OFF_VALUES=LOW,LOW
-DLORAWAN_RFSWITCH_RX_VALUES=HIGH,LOW
-DLORAWAN_RFSWITCH_RFO_LP_VALUES=LOW,HIGH
-DLORAWAN_RFSWITCH_RFO_HP_VALUES=LOW,HIGH
```

**Reply by ebelouet** [reply]
thank, I will try this
Eric

**Reply by fpistm** [reply]
Note that this is the value from the RAK variant, up to you to check the correct one for your board.
